### ISSUE TITLE
windows serverMiddleware HMR edge case 

### ISSUE THREAD

**Issue by usb248** [nuxt/nuxt.js#7097]
### Version

[v2.12.0](https://github.com/nuxt.js/releases/tag/v2.12.0)

### Reproduction link

[https://jsbin.com/](https://jsbin.com/)

### Steps to reproduce

npm run dev, then, 
Modify a servermiddleware file

### What is expected ?

No script crash

### What is actually happening?

D:\wamp64\www\...\node_modules\@nuxt\builder\dist\builder.js:6160
        for (const entry of dep2Entry[fileName]) {
                                     ^

TypeError: dep2Entry[fileName] is not iterable
TypeError: dep2Entry[fileName] is not iterable
    at FSWatcher.createFileWatcher.debounce_1 (D:\wamp64\www\...\node_modules\@nuxt\builder\dist\builder.js:6160:38)
    at invokeFunc (D:\wamp64\www\...\node_modules\@nuxt\builder\dist\builder.js:411:19)
    at trailingEdge (D:\wamp64\www\...\node_modules\@nuxt\builder\dist\builder.js:460:14)
    at Timeout.timerExpired [as _onTimeout] (D:\wamp64\www\...\node_modules\@nuxt\builder\dist\builder.js:448:14)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ....@1.0.0 dev: `nuxt`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ....@1.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\....-PC\AppData\Roaming\npm-cache\_logs\2020-03-17T19_08_41_177Z-debug.log

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10404">#c10404</a>)</em></sub></div>

**Comment by pi0** [comment]
Hi, @usb248 thanks for reporting the issue. But without repro, it is impossible to debug. Would you please share nuxt config or more details?

**Comment by usb248** [comment]
This evening no, tommorow. But i can say that i have:
```
serverMiddleware: [
...
    '~/api'
  ],
```
An index.js and others directories inside `api`directory.

**Comment by pi0** [comment]
Hi, @usb248 I tried using provided parts but could not reproduce maybe because something special is inside server middleware or API.

I added a quick fix commit (4b34941df802feed93048db6429f5d4eb836f942) to ensuring entry is in the graph but seems the root cause is an entry point is not being tracked.

If you had time, would you please to debug this? Either if you can give me (privately) access to project with this problem or yourself can start by editing `D:\wamp64\www...\node_modules@nuxt\builder\dist\builder.js`, search for `debounce((event, fileName)` and log what file is missing from graph like this:

```diff
      debounce((event, fileName) => {
++        if (!dep2Entry[fileName]) {
++          console.warn('HMR Entrypoint is missing but reloaded: ' + fileName)
++          return
++        }
        for (const entry of dep2Entry[fileName]) {
```

**Comment by usb248** [comment]
yes. index.js in api folder seems to cause the problem...
![image](https://user-images.githubusercontent.com/7659927/76994334-82243d00-694e-11ea-808d-8a7ff28015f7.png)

**Comment by pi0** [comment]
@usb248 My most guess is that it is a windows issue. Will try on windows machine :)

**Comment by usb248** [comment]
let me know if this is the case.
Thanks :)

**Comment by MLDMoritz** [comment]
Probably windows screws up filePaths. For a temporary solutions, may install linux subsystem on windows (Will speed up dev too ;-))

**Comment by pi0** [comment]
Update:

This issue is happening for windows users that have a serverMiddleware registered with implicit `index.js`

- 2.12.1 released with a fix that just prevents this error but HMR is not working for windows
- #7131 should fix this issue
- The current workaround is to change `~/api` to `~/api/index.js`

**Comment by pi0** [comment]
Hi again. It should be fixed on 2.12.2 :)
